DOC-4185 Document manual bootstrapping (#746)
corda/corda-docs-portal: content/en/platform/corda/5.0-alpha/deploying/deployment-tutorials/manual.md

Alternatively, the Corda CLI can be used to generate a shell script which may then be reviewed before executing against the broker.
The shell script makes use of the `kafka-topic.sh` script provided with a Kafka installation.

The Corda CLI command looks as follows:

```shell
corda-cli.sh topic -b <BOOTSTRAP-SERVERS> -k config.properties \
  create -r <RELICAS> -p <PARTITIONS> script -f <FILE> -c <CONCURRENCY>
```

Where `<FILE>` is the name of the file in which to save the script and `<CONCURRENCY>` is the number of topics to create in parallel to speed execution.

For example:

```shell
corda-cli.sh topic -b kafka-1.example.com -k config.properties \
  create -r 3 -p 10 script -f create.sh -c 6
```

The script `create.sh` would then be executed to created the topics.

## Database

By default, a Corda installation automatically creates and populates the database schema it requires.
This behavior can be disabled by setting the following override in the deployment configuration:

```yaml
bootstrap:
  db:
    enabled: false
```

The following steps can then be performed to manually set up the database prior to Corda installation.

### Create the Database Schemas

> The following Corda CLI command generates DML files in the directory `/tmp/db` for creating the database schema:

Reviewer: ```suggestion
2. Use the Corda CLI to generate DML files for creating the database schema. For example, the following command generates the files in the directory `/tmp/db`:
```

```shell
corda-cli.sh database spec -c -l /tmp/db
```

The DML generated should be reviewed and then executed against the database.

### Create RBAC Database Connection Configuration

> The following Corda CLI command generates DDL for populating the RBAC database connection configuration:

Reviewer: ```suggestion
4. Execute the following Corda CLI command to generate DDL for populating the RBAC database connection configuration:
```

```shell
corda-cli.sh initial-config create-db-config -u <RBAC-USERNAME> -p <RBAC-PASSWORD> \
  --name corda-rbac --jdbcURL jdbc:postgresql://<DB-HOST>:<DB-PORT>/<DB=NAME> \
  --jdbcPoolMaxSize <POOL-SIZE> --salt <SALT> --passphrase <PASSPHRASE> -l /tmp/db
```

The `<SALT>` and `<PASSPHRASE>` are used to encrypt the credentials in the database.
These must match the values specified in the Corda deployment configuration for the DB worker:

```yaml
workers:
  db:
    salt: <SALT>
    passphrase: <PASSPHRASE>
```

> An example usage might look as follows:

Reviewer: ```suggestion
For example:
```

```shell
corda-cli.sh initial-config create-db-config -u rbac-user -p rc9VLHU3 \
  --name corda-rbac --jdbcURL jdbc:postgresql://postgres.example.com:5432/cordacluster \
  --jdbcPoolMaxSize 5 --salt X3UaCpUH --passphrase UUWLhD8S -l /tmp/db
```

> The DDL generated should be reviewed and then executed against the database.

Reviewer: ```suggestion
5. Review the DDL files generated and then execute against the database.
```

> ### Create Crypto Database Connection Configuration

Reviewer: ```suggestion

```

> The following Corda CLI command generates DDL for populating the Crypto database connection configuration:

Reviewer: ```suggestion
6. Execute the following Corda CLI command to generate DDL for populating the Crypto database connection configuration:
```

```shell
corda-cli.sh initial-config create-db-config -u <CRYPTO-USERNAME> -p <CRYPTO-PASSWORD> \
  --name corda-crypto --jdbcURL jdbc:postgresql://<DB-HOST>:<DB-PORT>/<DB=NAME>?currentSchema=CRYPTO \
  --jdbcPoolMaxSize <POOL-SIZE> --salt <SALT> --passphrase <PASSPHRASE> -l /tmp/db
```

The `<SALT>` and `<PASSPHRASE>` must match those used above and specified in the Corda deployment configuration.

> An example usage might look as follows:

Reviewer: ```suggestion
For example:
```

```shell
corda-cli.sh initial-config create-db-config -u crypto-user -p TqoCp4v2 \
  --name corda-crypto --jdbcURL jdbc:postgresql://postgres.example.com:5432/cordacluster?currentSchema=CRYPTO \
  --jdbcPoolMaxSize 5 --salt X3UaCpUH --passphrase UUWLhD8S -l /tmp/db
```

> The DDL generated should be reviewed and then executed against the database.

Reviewer: ```suggestion
7. Review the DDL files generated and then execute against the database.
```

> ### Create the Initial Admin User

Reviewer: ```suggestion

```

> The following Corda CLI command generates DDL for populating the initial admin user for Corda:

Reviewer: ```suggestion
8.Execute the following Corda CLI command to generate DDL for populating the initial admin user for Corda:
```

```shell
corda-cli.sh initial-config create-user-config -u <INITIAL-ADMIN-USERNAME> -p <INITIAL-ADMIN-PASSWORD> -l /tmp/db
```

> The DDL generated should be reviewed and then executed against the database.

Reviewer: ```suggestion
9. Review the DDL files generated and then execute against the database.
```

> ### Create the Database Users and Grant Access

Reviewer: ```suggestion

```

> Create the RBAC and Crypto users and grant access as follows:

Reviewer: ```suggestion
10. Create the RBAC and Crypto users and grant access as follows:
```

```sql
CREATE USER <RBAC-USERNAME> WITH ENCRYPTED PASSWORD '<RBAC-PASSWORD>';
GRANT USAGE ON SCHEMA RPC_RBAC to <RBAC-USERNAME>;
GRANT SELECT, INSERT, UPDATE, DELETE ON ALL TABLES IN SCHEMA RPC_RBAC to <RBAC-USERNAME>;
CREATE USER <CRYPTO-USERNAME> WITH ENCRYPTED PASSWORD '<CRYPTO-PASSWORD>';
GRANT USAGE ON SCHEMA CRYPTO to <CRYPTO-USERNAME>;
GRANT SELECT, INSERT, UPDATE, DELETE ON ALL TABLES IN SCHEMA CRYPTO to <CRYPTO-USERNAME>;
```

> ### Create the Initial Crypto Configuration

Reviewer: ```suggestion

```

> The following Corda CLI command generates DDL for populating the initial crypto configuration:

Reviewer: ```suggestion
11. Execute the following Corda CLI command to generate DDL for populating the initial crypto configuration:
```

```shell
corda-cli.sh initial-config create-crypto-config --salt <SALT> --passphrase <PASSPHRASE> -l /tmp/db
```

The `<SALT>` and `<PASSPHRASE>` must match those used above and specified in the Corda deployment configuration.

> The DDL generated should be reviewed and then executed against the database.

Reviewer: ```suggestion
12. Review the DDL files generated and then execute against the database.
```

## RBAC Roles

A post-install job normally creates three default RBAC roles for the Corda API.

> Automatic creation of these RBAC roles can be disabled as follows:

Reviewer: ```suggestion
1. Set the following override in the deployment configuration to disable the automatic creation:
```

```yaml
bootstrap:
  rbac:
    enabled: false
```

> To manually create the three RBAC roles, execute the following three commands:

Reviewer: ```suggestion
2. Execute the following three commands:
```
